Azure.ResourceManager.Core/Resources/Identity.cs

```diff
@@ -1,32 +1,203 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using Azure.Core;
 using System;
+using System.Collections.Generic;
+using System.Text.Json;
 
 namespace Azure.ResourceManager.Core
 {
     /// <summary>
     ///     Represents a managed identity
     ///     TODO: fill in properties, implement comparison and equality methods and operator overloads
     /// </summary>
-    public class Identity : IEquatable<Identity>, IComparable<Identity>
+    public class Identity
     {
-        public Guid TenantId { get; set; }
+        private const string SystemAssigned = "SystemAssigned";
+        private const string UserAssigned = "UserAssigned";
+        private const string SystemAndUserAssigned = "SystemAssigned, UserAssigned";
 
-        public Guid PrincipalId { get; set; }
+        public SystemAssignedIdentity SystemAssignedIdentity { get; set; } // Need to decide on setter
 
-        public Guid ClientId { get; set; }
+        public IDictionary<ResourceIdentifier, UserAssignedIdentity> UserAssignedIdentities { get; set; } // maintain structure of {id, (clientid, principal id)} in case of multiple UserIdentities
 
-        public ResourceIdentifier ResourceId { get; set; }
+        public Identity() : this(null, false) { } //not system or user
 
-        public int CompareTo(Identity other)
+        public Identity(Dictionary<ResourceIdentifier, UserAssignedIdentity> user, bool useSystemAssigned)
         {
-            throw new NotImplementedException();
+            // check for combination of user and system on the impact to type value
+            SystemAssignedIdentity = useSystemAssigned ? new SystemAssignedIdentity() : null;
+            UserAssignedIdentities = new Dictionary<ResourceIdentifier, UserAssignedIdentity>();
+            if (user != null)
+            {
+                foreach (KeyValuePair<ResourceIdentifier, UserAssignedIdentity> id in user)
+                {
+                    UserAssignedIdentities.Add(id.Key, id.Value);
+                }
+            }
+        }
+
+        public Identity(SystemAssignedIdentity systemAssigned, IDictionary<ResourceIdentifier, UserAssignedIdentity> user)
+        {
+            SystemAssignedIdentity = systemAssigned;
+            if (user == null)
+            {
+                UserAssignedIdentities = new Dictionary<ResourceIdentifier, UserAssignedIdentity>();
+            }
+            else
+            {
+                UserAssignedIdentities = user;
+            }
         }
 
         public bool Equals(Identity other)
         {
-            throw new NotImplementedException();
+            if (other == null)
+                return false;
+
+            bool isUserAssignedEqual = true;
```

Comment:
We don't need to track this here.  If this ever turns false during the loop you can safely return false.  Since this value will be anded with SystemAssignedIdentity.Equals and if its false it doesn't matter what the result of SystemAssignedIdentity.Equals is